pandas/tests/io/json/test_pandas.py

```diff
@@ -1032,6 +1032,67 @@ def test_to_jsonl(self):
         assert result == expected
         assert_frame_equal(pd.read_json(result, lines=True), df)
 
+    def test_read_jsonchunks(self):
+        # GH17048: memory usage when lines=True
+        df = pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]})
+
+        def get_strio():
+            return StringIO(df.to_json(lines=True, orient="records"))
+
+        def test_with_chunksize(c):
+            iterator = pd.read_json(get_strio(), lines=True, chunksize=c)
+            return pd.concat(iterator)
+
+        unchunked = pd.read_json(get_strio(), lines=True)
+
+        chunked = test_with_chunksize(1)
+
+        assert_frame_equal(chunked, unchunked)
+
+        chunked_float = test_with_chunksize(1.0)
+        assert_frame_equal(chunked_float, unchunked)
+
+        msg = r"'chunksize' must be an integer >=1"
+
+        with tm.assert_raises_regex(ValueError, msg):
+            test_with_chunksize(0)
+
+        with tm.assert_raises_regex(ValueError, msg):
+            test_with_chunksize(-1)
+
+        with tm.assert_raises_regex(ValueError, msg):
+            test_with_chunksize(-2.2)
+
```

Comment:
rather than making a very long test, pls break into several tests, e.g. errors can be a test

Reply:
Will do, thanks for the suggestion.